lib/AST/Module.cpp

```diff
@@ -276,8 +276,11 @@ void SourceLookupCache::populateMemberCache(const ModuleDecl &Mod) {
                              "populate-module-class-member-cache");
 
   for (const FileUnit *file : Mod.getFiles()) {
-    auto &SF = *cast<SourceFile>(file);
-    addToMemberCache(SF.getTopLevelDecls());
+    assert(isa<SourceFile>(file) ||
+           isa<SynthesizedFileUnit>(file));
+    SmallVector<Decl *, 8> decls;
+    file->getTopLevelDecls(decls);
+    addToMemberCache(decls);
```

Comment:
@slavapestov Do you think it's relevant for `SourceLookupCache::populateMemberCache` to cache the decls from `SynthesizedFileUnit` that are part of a module composed otherwise only of `SourceFiles`? The differentiation and actors logic appears to insert `SynthesizedFileUnits` in such modules.

Reply:
If unqualified lookup can see those decls, I believe they must be present in the cache (since there is no other mechanism to find them), unless we bypass the cache entirely and check each FileUnit in that case (but that would be bad for performance).

Reply:
Alright, thanks!